aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandlerTest.java

```diff
@@ -81,18 +88,82 @@
+                .channels(Channel.builder()
+                        .uri("channelUri")
+                        .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                        .build())
+                .build();
 
         @BeforeEach
         public void setup() {
-            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).build());
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName)
+                    .anomalyDetectionNotificationConfiguration(Collections.singletonList(software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                            .channelUri("channelUri")
+                            .build()))
+                    .build());
 
             doReturn(CreateProfilingGroupResponse.builder()
-                         .profilingGroup(ProfilingGroupDescription.builder().name(profilingGroupName).build())
-                         .build())
+                        .profilingGroup(ProfilingGroupDescription.builder().name(profilingGroupName).build())
+                        .build())
                     .when(proxy).injectCredentialsAndInvokeV2(eq(createPgRequest), any());
         }
 
+        @Test
+        public void itCreatesNotificationChannelSuccess() {
+            subject.handleRequest(proxy, request, null, logger);
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(createPgRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any());
+            verifyNoMoreInteractions(proxy);
+        }
+
+        @Test
+        public void itDeletesProfilingGroupWhenAddNotificationChannelFails() {
+            when(proxy.injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any())).thenThrow(InternalServerException.class);
+            assertThrows(CfnServiceInternalErrorException.class,
+                    () -> subject.handleRequest(proxy, request, null, logger));
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(createPgRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(deleteProfilingGroupRequest), any());
+            verifyNoMoreInteractions(proxy);
+        }
+
+        @Test
+        public void itCreatesNotificationChannelWithOptionalChannelIdNotSetSuccess() {
+            doReturn(AddNotificationChannelsResponse.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any());
+
+            final ProgressEvent<ResourceModel, CallbackContext> response = subject.handleRequest(proxy, request, null, logger);
+            assertSuccessfulResponse(response);
+        }
+
+        @Test
+        public void itCreatesNotificationChannelWithOptionalChannelIdSetSuccess() {
```

Comment:
Minor: Suggested small simplification to names:

* `itCreatesNotificationChannelWithOptionalChannelIdNotSetSuccess` => `itSucceedsWhenChannelIdIsMissing`
* `itCreatesNotificationChannelWithOptionalChannelIdSetSuccess` => `itSucceedsWhenChannelIdExists`

Reply:
:point_up_2:  This one was resolved but has not been addressed.

Reply:
I'm so confused... I think a lot of my changes have been reverted somehow.  Let me readd